### ISSUE TITLE
InvalidReturnType is emitted for phpdoc class with wrong casing, but classes are case insensitive

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#619]
Namespaces are also case insensitive

This bug goes a while back, versions from 0.3.9 to 1.0.5 are affected (I don't think 0.3.8 emitted invalidreturntype for those methods, even if there was an invalid return type)

Observed:

```
src/casesensitive.php:6: [E0001] InvalidReturnType: The declared return type 'array<int, example>' for MyNS\Example::test is incorrect, got 'array{0:MyNS\Example}' (column 9)
src/casesensitive.php:8: [E0001] InvalidReturnStatement: The type 'array{0:MyNS\Example}' does not match the declared return type 'array<int, example>' for MyNS\Example::test (column 9)
src/casesensitive.php:11: [E0001] UndefinedClass: Class or interface example does not exist (column 17)
src/casesensitive.php:11: [E0001] InvalidReturnType: The declared return type 'example' for MyNS\Example::instance is incorrect, got 'MyNS\Example' (column 17)
src/casesensitive.php:13: [E0001] InvalidReturnStatement: The type 'MyNS\Example' does not match the declared return type 'example' for MyNS\Example::instance (column 9)
```

Expected: Should not warn. Optionally, could warn about the case being different from the class declaration

Also, the way that the error message type is displayed/inferred seems odd. Why is it `array<int,example>` instead of `array<int,MyNS\example>`

```php
<?php

namespace MyNS;

class Example {
    /** @return array<int,example> */
    public static function test() : array {
        return [new Example()];
    }

    /** @return example */
    public static function instance() {
        return new Example();
    }
}
```

**Comment by muglug** [comment]
Thanks!
